src/main/webapp/get-blur-areas.js

```diff
@@ -0,0 +1,264 @@
+                // are no greater than the image
+                const imageObject = new Image();
+                const imageUrl = URL.createObjectURL(image);
+
+                imageObject.onload = function() {
+                  const blurAreas = [];
+
+                  for (const jsonRect of jsonBlurAreas) {
+                    let rect;
+                    try {
+                      rect = new Rect(jsonRect, imageObject);
+                    } catch (error) {
+                      // rectangle is invalid - skip it
+                      console.log('Invalid rectangle : ' + error.message,
+                          jsonRect);
+                      continue;
+                    }
+
+                    blurAreas.push(rect);
+                  }
+
+                  if (blurAreas.length === 0) {
+                    alert('We did not recognize any areas to blur :(');
+                  }
+
+                  resolve(blurAreas);
+                };
+
+                imageObject.src = imageUrl;
+              },
+
+              (error) => {
+                if (error.message !== response.status + ' server error') {
+                  alert('Unknown error while parsing results.' +
+                      '\nPlease reload the page and try again ' +
+                      'after some time.');
+                  reject(error);
+                }
+              },
+          );
+        },
+
+        (error) => {
+          alert('Unknown error while uploading image.' +
+              '\nPlease reload the page and try again after some time.');
+          reject(error);
+        },
+    );
+  });
+}
+
+/**
+ * Constructor for rectangle to blur.
+ * Takes as parameter rectangle from server - array with 4 points,
+ * point is an object with properties 'x' and 'y'.
+ * Validates rectangle from server.
+ * @param {Array<Object>} rect
+ * @param {Image} image
+ * @constructor
+ */
+function Rect(rect, image) {
+  if (!Array.isArray(rect)) {
+    throw new Error('Object passed here is not an Array. ' +
+        'It must be an Array of points');
+  }
+
+  // has 4 points
+  if (rect.length !== 4) {
+    throw new Error(`Rectangle object must contain exactly 4 corner ` +
+        `points. This rectangle has ${rect.length} points.`);
+  }
+
+  // points must have x and y properties
+  for (const point of rect) {
+    if (!point.hasOwnProperty('x')) {
+      throw new Error(`Point ${JSON.stringify(point)} ` +
+      `does not have "x" property`);
+    }
+    if (!point.hasOwnProperty('y')) {
+      throw new Error(`Point ${JSON.stringify(point)} ` +
+          `does not have "y" property`);
+    }
+  }
+
+  // points must not duplicate
+  for (let i = 0; i < 4; i++) {
+    for (let j = 0; j < i; j++) {
+      if (rect[i].x === rect[j].x && rect[i].y === rect[j].y) {
+        throw new Error(`Duplicate points ${i} : (${rect[i].x}, ${rect[i].y})` +
+            ` and ${j} : (${rect[j].x}, ${rect[j].y})`);
```

Comment:
Nit (feel free to ignore): rect[i] and rect[j] are the same, so you don't need to print them both.